docs/service-implementation-101.md

```diff
@@ -0,0 +1,390 @@
+# Service Implementation 101
+> valid for: 2022-09-02
+
+Are you hoping to learn how to implement a new service for our existing capabilities (i.e., kv, lockd, mq, or pubsub)? You have come to the right place!
+
+> Note: If you would like to develop a new capability outside of our provided interfaces, you will first have to create its' WIT. Please, create an issue to start a design discussion, and make a PR with the `.wit` file itself prior to its' `slight` implementation.
+
+The DeisLabs Engineering Team has designed each of its' main capabilities to be easily extensible, and some macros to take care of boiler-plate code for you. In this tutorial, we will implement a new `pubsub` implementation — a local one with Mosquitto!
+
+## Getting Started
+
+Assuming that you are on the root of the SpiderLightning repository that you locally cloned, to get started, do:
+
+```sh
+git branch my_local_pubsub_implementation c73826a1d522bc3e1d2f0387481880edecd3e3d3
```

Comment:
```suggestion
git checkout -b my_local_pubsub_implementation
```

Reply:
I want this tutorial to work regardless of the state of the main branch~ That's why I'm branching off of a specific commit hash.

Reply:
Won't that create a large PR when targeting main, or cause the developer to need to rebase to get started?